src/update_agent/actor.rs

```
@@ -478,18 +478,20 @@ impl UpdateAgentInfo {
         })?;
 
         if pending.version != release.version {
-            bail!(
+            log::error!(
                 "expected pending deployment '{}', but found '{}' instead",
                 release.version,
                 pending.version
             );
+            return Ok(false);
         }
         if pending.checksum != release.checksum {
-            bail!(
+            log::error!(
                 "detected checksum mismatch for pending deployment '{}', got unexpected value '{}'",
                 release.version,
                 release.checksum,
             );
+            return Ok(false);
```

Hmm, why do we need this?

---

The rest of the function was built that way, and the bail error text was swallowed so I just aligned all the checks to use the same syntax

---

This changes the semantics of the function. I think those differences might be on purpose.

In the OSTree case, if the checksum doesn't match, then something went really wrong because we expicitly deployed by checksum. Whereas the stream check below this line is done here because it can't be done before deploying; it's part of the commit metadata. (Though I guess we could pull it before deploying.)

We can't exactly transpose this to the OCI case, because there while we do deploy by digest, that digest doesn't actually mean much within the object store. That said, here I think `checksum` is actually the pullspec in the OCI path as this PR is written (right?), so we do also expect it to be always true. It's basically checking that the custom origin URL we've set has correctly been set. I think then we do still want to keep this a hard error?

For the version check, I think the change to not make it a hard error makes sense.

---

Reverted this as per discsussion